pkg/cloud/aws/actuators/scope.go

```diff
@@ -131,40 +133,45 @@ func (s *Scope) Region() string {
 	return s.ClusterConfig.Region
 }
 
-func (s *Scope) storeClusterConfig(cluster *clusterv1.Cluster) (*clusterv1.Cluster, error) {
-	ext, err := v1alpha1.EncodeClusterSpec(s.ClusterConfig)
-	if err != nil {
-		return nil, err
-	}
-
-	cluster.Spec.ProviderSpec.Value = ext
-	return s.ClusterClient.Update(cluster)
-}
-
-func (s *Scope) storeClusterStatus(cluster *clusterv1.Cluster) (*clusterv1.Cluster, error) {
-	ext, err := v1alpha1.EncodeClusterStatus(s.ClusterStatus)
-	if err != nil {
-		return nil, err
-	}
-
-	cluster.Status.ProviderStatus = ext
-	return s.ClusterClient.UpdateStatus(cluster)
-}
-
 // Close closes the current scope persisting the cluster configuration and status.
 func (s *Scope) Close() {
 	if s.ClusterClient == nil {
 		return
 	}
-
-	latestCluster, err := s.storeClusterConfig(s.Cluster)
+	ext, err := v1alpha1.EncodeClusterSpec(s.ClusterConfig)
 	if err != nil {
-		s.Error(err, "failed to store provider config")
+		s.Error(err, "failed encoding cluster spec", "cluster-name", s.Cluster.Name)
 		return
 	}
-
-	_, err = s.storeClusterStatus(latestCluster)
+	status, err := v1alpha1.EncodeClusterStatus(s.ClusterStatus)
 	if err != nil {
-		s.Error(err, "failed to store provider status")
+		s.Error(err, "failed encoding cluster status", "cluster-name", s.Cluster.Name)
+		return
+	}
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		s.V(2).Info("updating cluster", "cluster-name", s.Cluster.Name, "cluster-resource-version", s.Cluster.ResourceVersion)
+		s.Cluster.Spec.ProviderSpec.Value = ext
+		s.V(6).Info("cluster status before update", "cluster-status", s.Cluster.Status)
```

Comment:
For my info: how do I make these logs show up? is it the `-v=#` command line flag?

Reply:
that is correct

Reply:
In the generated `provider-components.yaml` find the cluster-api-provider-aws controller and add a `-v=#` to the argument list